msu/hooks/scenarios/scenario_manager.nut

```diff
@@ -1,11 +0,0 @@
-::mods_hookNewObjectOnce("scenarios/scenario_manager", function(o) {
-	foreach (scenario in o.m.Scenarios)
-	{
-		local onUpdateLevel = ::mods_getMember(scenario, "onUpdateLevel");
-		scenario.onUpdateLevel <- function( _bro )
-		{
-			onUpdateLevel(_bro);
-			_bro.getSkills().onUpdateLevel();
-		}
-	}
-});
```

Comment:
This was a ghetto fix when we made it originally, coz we had no way of hooking leaves of `starting_scenario`. So, we instead hooked `scenario_manager` and hooked the scenarios on runtime. This is not good. We should now use `hookTree` of Modern Hooks as it is meant for use cases like this.

Reply:
This also means that we can probably get rid of the Boolean we use in `skill_container` to prevent recursive calls to `onUpdateLevel`

Reply:
> This was a ghetto fix when we made it originally, coz we had no way of hooking leaves of starting_scenario. So, we instead hooked scenario_manager and hooked the scenarios on runtime. This is not good. We should now use hookTree of Modern Hooks as it is meant for use cases like this.

I see, fair enough

> This also means that we can probably get rid of the Boolean we use in skill_container to prevent recursive calls to onUpdateLevel

That I leave for you to determine